src/setMiddleC.js

```diff
@@ -26,20 +27,22 @@ function transposeOctave(initialOctave) {
 /**
  * Takes a noteObj and transposes the note into the octave given by transposition 
  * @param {String/Array} noteArg		The Array/String contaning the note(s)
+ * @param {Integer} octave Optional octave to transpose to  
  * @return {String(s)} 	The correctly transposed note(s)
  */
-function transposeNote(noteArg) {
+const transposeNote = (noteArg, octave) => {
     let note;
 	assert(noteArg !== undefined && (typeof noteArg == 'string' || typeof noteArg == 'object') && noteArg.note === undefined && noteArg[0] !== undefined, 'NoteArg must contain a note that is either an array or a string.');
+	assert(Number.isInteger(octave) || octave===undefined, 'Octave must be an integer');
```

Comment:
Also in the previous line can we stick to the convention of triple equal to..  for consistency